frontend/src/features/business/components/VendorDetails.tsx

```diff
@@ -57,7 +58,7 @@ export function VendorDetails({ vendor }: VendorDetailsProps) {
         });
       }
     };
-  }, []);
+  }, [vendor.slug]);
```

Comment:
this dependency is needed if theoretically you could navigate from one vendor detail page to another one using next/link navigation?